src/main/webapp/script.js

```diff
@@ -2,10 +2,92 @@ function getRandomRestaurant() {
   const restaurants =
       ['Din Tai Fung', 'Mcdonalds', 'KFC', 'Subway'];
 
-  // Pick a random greeting.
+  // Pick a random restaurant.
   const restaurant = restaurants[Math.floor(Math.random() * restaurants.length)];
 
   // Add it to the page.
-  const resElemenet = document.getElementById('restaurant-name');
-  resElemenet.innerText = restaurant;
+  const resElement = document.getElementById('restaurant-name');
+  resElement.innerText = restaurant;
+}
+
+function initMap() {
+  var map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: 1.2839, lng: 103.8607}, // default location, marina bay
+    zoom: 13
+  });
+  var card = document.getElementById('pac-card');
+  var input = document.getElementById('pac-input');
+
+  map.controls[google.maps.ControlPosition.TOP_RIGHT].push(card);
+
+  var autocomplete = new google.maps.places.Autocomplete(input);
+
+  // Set the data fields to return when the user selects a place.
+  autocomplete.setFields(
+      ['address_components', 'geometry', 'icon', 'name']);
+
+  var infowindow = new google.maps.InfoWindow();
+  var infowindowContent = document.getElementById('infowindow-content');
+  infowindow.setContent(infowindowContent);
+  var marker = new google.maps.Marker({
+    map: map,
+    anchorPoint: new google.maps.Point(0, -29)
+  });
+
+  // If user selects one of the locations in the autocomplete UI
+  autocomplete.addListener('place_changed', function() {
+    infowindow.close();
+    marker.setVisible(false);
+    var place = autocomplete.getPlace();
+    if (!place.geometry) {
+      // User entered the name of a Place that was not suggested and
+      // pressed the Enter key, or the Place Details request failed.
+      window.alert("No details available for input: '" + place.name + "'");
+      return;
+    }
+
+    // If the place has a location, then update map.
+    if (place.geometry.viewport) {
+      map.fitBounds(place.geometry.viewport);
+    } else {
+      map.setCenter(place.geometry.location);
+      map.setZoom(17); // Apparently it looks better
+    }
+    marker.setPosition(place.geometry.location);
+    marker.setVisible(true);
+
+    var address = '';
+    if (place.address_components) {
+      address = [
+        (place.address_components[0] && place.address_components[0].short_name || ''),
+        (place.address_components[1] && place.address_components[1].short_name || ''),
+        (place.address_components[2] && place.address_components[2].short_name || '')
+      ].join(' ');
+    }
+
+    infowindowContent.children['place-icon'].src = place.icon;
+    infowindowContent.children['place-name'].textContent = place.name;
+    infowindowContent.children['place-address'].textContent = address;
+    infowindow.open(map, marker);
+  });
+
+
+  document.getElementById("submit-button").addEventListener("click", function(){
```

Comment:
Planning to add further expansions such as sending the location to the servlet within this function